Source/AwsCommonRuntimeKit/http/HTTPStreamCallbackCore.swift

```diff
@@ -105,7 +110,11 @@ private func onComplete(stream: UnsafeMutablePointer<aws_http_stream>?,
     let httpStreamCbData = Unmanaged<HTTPStreamCallbackCore>.fromOpaque(userData).takeUnretainedValue()
     let onStreamCompleteFn = httpStreamCbData.requestOptions.onStreamComplete
     guard errorCode == AWS_OP_SUCCESS else {
-        onStreamCompleteFn(.failure(CommonRunTimeError.crtError(CRTError(code: errorCode))))
+        do {
+            try onStreamCompleteFn(.failure(CommonRunTimeError.crtError(CRTError(code: errorCode))))
+        } catch {
+            fatalError("C callback needs to updated.")
```

Comment:
@waahm7  is there a way we cal close the connection here?

Reply:
While we might be able to handle this error at this location, it is not the best place to do so. Ideally, the SDK should have a pointer to the stream in this function and should send the appropriate [RST_STREAM](https://datatracker.ietf.org/doc/html/rfc9113#section-6.4) or [GO_AWAY](https://datatracker.ietf.org/doc/html/rfc9113#name-goaway) frame for an HTTP/2 connection.

Reply:
Discussed offline that we will not be making onComplete throwable.